tsl/src/nodes/skip_scan/planner.c

```diff
@@ -625,9 +710,30 @@ skip_scan_path_create(PlannerInfo *root, IndexPath *index_path, DistinctPathInfo
 	 * to (ndistinct / rows) * total leading to SkipScan not being chosen for queries on
 	 * hypertables with a lot of excluded chunks.
 	 */
+	double startup = (do_decompress_cost ?
+						  /* If we need to scan decompressed batch to find a tuple passing the qual,
+						   * we assume the GUC-specified fraction of the batch
+						   * needs to be scanned for each distinct value (0.5 by default).
+						   * Startup cost should be capped by at least startup cost of the index
+						   * scan if ratio is too small.
+						   */
+						  Max(child_path->startup_cost * ts_guc_compression_batch_qual_match_ratio,
+							  index_path->path.startup_cost) :
+						  index_path->path.startup_cost);
+	double total = child_path->total_cost;
+	double rows = index_path->path.rows;
+	double uncompressed_rows = child_path->rows;
+
 	skip_scan_path->cpath.path.startup_cost = startup;
```

Comment:
```suggestion
	skip_scan_path->cpath.path.startup_cost = child_path->startup_cost;
```

This is becoming confusing, so let's focus on the obviously incorrect things. The startup cost can't be less than the startup cost of the child plans. The SkipScan itself requires minimal setup, so let's just set the startup cost the same as child startup cost.

Reply:
Agree. I plan to redo it as described here, which should address this issue too: https://github.com/timescale/timescaledb/pull/7983#discussion_r2067665721